types/qaengine/problem.go

```diff
@@ -69,14 +71,28 @@ func NewProblem(p *qagrpc.Problem) (prob Problem, err error) {
 		logrus.Errorf("Unable to convert defaults : %s", err)
 		return prob, err
 	}
-	return Problem{
+	pp := Problem{
 		ID:      p.Id,
 		Type:    SolutionFormType(p.Type),
 		Desc:    p.Description,
 		Hints:   p.Hints,
 		Options: p.Options,
 		Default: defaults,
-	}, nil
+	}
+	if p.Pattern != "" {
+		pp.Validation = func(ans interface{}) error {
+			reg, err := regexp.Compile(p.Pattern)
+			if err != nil {
+				return err
+			}
+			a := ans.(string)
```

Comment:
if the answer type is not string, which is the case with question of type like multiselect, etc, this will panic, isn't it?

Also, don't we want validation for other types of questions too?